prometheus_juju_exporter/collector.py

```diff
@@ -128,6 +121,21 @@ def _get_labels_to_remove(self, gauge_name: str) -> None:
         for label in stale_labels.values():
             self.data[gauge_name]["labelvalues_remove"].append(list(label[0].values()))
 
+    def _get_machine_type(self, machine: Dict) -> MachineType:
```

Comment:
not-blocking: This function is not tested directly with any unit test.

Reply:
I never heart that private function should not be tested directly. Interacting with private functions outside of python file or class should not be used, but this certainly does not apply to unit tests. Each unit test should test only functionality of one function, so each other function should be mocked with expected output.

Reply:
I tend to agree. I liked your previous test_get_machine_type function because it's smaller and clearer. test_get_stats may cover the same lines of code but it's not as obvious.

Reply:
There are many articles laying out the reasons not to test private methods directly ([example](https://anthonysciamanna.com/2016/02/14/should-private-methods-be-tested.html)). The main argument is that logic of private methods should be tested with public interfaces, simulating the their real usage. 

This being said, I'm don't have a strong opinion on this and I agree that it is cleaner to directly test the function in this case. I've reverted the changes.

Reply:
I read your example and searched more and I have to say that I see some sense why not to test private functions. However to my it will always make much more sense, at least in Python where mocking is easy and the distinction between private and public functions is not the same as in, for example in C++, to test each function individually and here is my reasons:
- [Readability] Test functions are shorter and more readable, because I don't have to try too many combinations in them. 
- [Easier to detect any changes] Mocking all private methods can be beneficial especially when you need to change functionality of any private method or functionality changed with some packages update. In this scenario, you will immediately see the test failure for the private function, so it is easy to identify why.
- [all inputs/outputs and failures] You can check all possible inputs/outputs and failures for each function. Many times, when as a developer I don't understand what certain functions do, I can understand them through good unit tests or get an idea of what to expect from the function.

As I was writing this, I realized one more thing, is it even a good idea to make the `_get_machine_type` function private?

Reply:
@agileshaw thanks for the article; I think the argument it makes is not so much that it's not necessary to test private methods, but rather that if you feel the need to test a private method then that method should probably be public. In fact upon re-reading the Collector class code I am starting to think that it tries to do too much and should be broken down into separate classes. That said, this sort of refactoring is out of scope for this PR so I would advise to keep it in a separate PR. This one can be merged once you re-add test_get_machine (even if it's testing a private method).

Reply:
Thanks @aieri and @rgildein for the discussion. I've already reverted the changes and was just trying to continue this valuable conversation. After more thinking to it, I think it would indeed be good to move the helper functions to a different class and change them to public functions. I will open another PR for refactoring.